HW4_smythe_full.py

```diff
@@ -0,0 +1,370 @@
+            xaxis_title=f"Variable: {feature_name}",
+            yaxis_title="y",
+        )
+        file_name = f"./HW4_{feature_name}_cat_cat_violin_plot.html"
+        fig.write_html(file=file_name, include_plotlyjs="cdn")
+        cat_cat_file_list += [f'<a href="file:///{file_name}">link</a>']
+    return cat_cat_file_list
+
+
+def plot_cat_cont(to_plot):
+    # Violin plot on response grouped by predictor
+    user_db = to_plot
+    X = user_db.data
+    y = user_db.target
+    cat_con_file_list = []
+    #
+    for idx, column in enumerate(X.T):
+        feature_name = user_db.feature_names[idx]
+        feature_name = feature_name.replace("/", "-")
+        predictor = sm.add_constant(column)
+        #
+        logistic_regression_model = LogisticRegression(random_state=4622).fit(
+            predictor, y
+        )
+        print(f"Variable: {feature_name}")
+        print(logistic_regression_model.score(predictor, y))
+        # Get the score
+        the_score = logistic_regression_model.score(predictor, y)
+        # Plot the Figure
+        fig = px.violin(x=column, y=y, color=column)
+        fig.update_layout(
+            title=f"Variable:{feature_name},Logistic Regression Fit Score={the_score}",
+            xaxis_title=f"Variable: {feature_name}",
+            yaxis_title="y",
+        )
+        file_name = f"./HW4_{feature_name}_cat_con_violin_plot{idx}.html"
+        fig.write_html(file=file_name, include_plotlyjs="cdn")
+        cat_con_file_list += [file_name]
+    return cat_con_file_list
+
+
+# add boolean check incase type check is wrong
+def col_is_bool(df, col):
+    c = len(df[col].unique())
+    if c == 2:
+        True
+    else:
+        False
+    return col
+
+
+# predictor select for any dataset
+def predictor_select(ds):
+    predictors = ds.feature_names
+    # index_list = [i+1 for i in range(len(predictors))]
+    selection_df = pd.DataFrame(predictors, columns=["Predictor"])
+    selection_df.index += 1
+    print(selection_df)
+    pred_choice = input(
+        "Which predictors will you use? (use index numbers separate by a comma) :"
+    )
+    pred_choice = pred_choice.split(",")
+    for i in range(0, len(pred_choice)):
+        pred_choice[i] = int(pred_choice[i])
+    feature_list = [selection_df.at[i, "Predictor"] for i in pred_choice]
+
+    return feature_list
+
+
+def print_heading(title):
+    print("*" * 80)
+    print(title)
+    print("*" * 80)
+    return
+
+
+def list_to_df(from_df, to_df, col):
+    temp_list = from_df[col].to_list()
+    to_df[col] = pd.Series(temp_list, index=to_df.index)
+
+
+# Main functions
+def main():
+    start_t = datetime.now()
+    # set built-in dataset choices
+    ds1 = datasets.load_boston()
+    ds2 = datasets.load_iris()
+    ds3 = datasets.load_diabetes()
+    ds4 = datasets.load_digits()
+    ds5 = datasets.load_linnerud()
+    ds6 = datasets.load_wine()
+    ds7 = datasets.load_breast_cancer()

```

Comment:
Some of these datasets worked for your code but some didn't. I think it was failing when I was trying to plot.